libsolidity/ast/AST_accept.h

```diff
@@ -403,7 +403,8 @@ void Block::accept(ASTVisitor& _visitor)
 void Block::accept(ASTConstVisitor& _visitor) const
 {
 	if (_visitor.visit(*this))
-		listAccept(m_statements, _visitor);
+		if (m_statements.size() > 0 && m_statements[0] != nullptr)
```

Comment:
I don't understand - this change should be done inside listAccept, as far as I can see. Why it is important that the first element is non-null?

Reply:
Here is what _was_ happening. Some contracts/functions were so bolixed that the contract was created but we could even create the first statement correctly - it was null. 

Since information about the contract or function like its name or signature is useful having such an entity with a null statement has some benefit. 

However, I just retested everything with all the bad contracts I have around and do not see a SEGV any more like I used to. Instead what I think is going on is that we just are creating no AST whatsoever in these situations. So we may encounter this in the future. But the future is future.

So in commit f333cebb5 this is now removed. The other change in the commit which is a small typo in error message formatting I'll put in as a separate PR. 

Also, I'll be adding a (small) test suite so that it is more clear when some test like this is needed, _why_ it is needed.